references/product-recommendations-api/pom.xml

```diff
@@ -0,0 +1,312 @@
+				<apigee.authtype>oauth</apigee.authtype>
+				<apigee.bearer>${bearer}</apigee.bearer>
+				<apigee.options>override</apigee.options>
+			</properties>
+		</profile>
+	</profiles>
+
+	<build>
+		<plugins>
+			<!-- copy the full apiproxy folder to target folder -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-resources-plugin</artifactId>
+				<version>3.1.0</version>
+				<executions>
+					<execution>
+						<id>copy-resources</id>
+						<phase>process-resources</phase>
+						<goals>
+							<goal>copy-resources</goal>
+						</goals>
+						<configuration>
+							<overwrite>true</overwrite>
+							<resources>
+								<resource>
+									<!--source -->
+									<directory>${project.root.dir}</directory>
+									<filtering>true</filtering>
+									<includes>
+										<include>edge.json</include>
+										<include>apiproxy/**</include>
+										<include>test/integration/**</include>
+									</includes>
+								</resource>
+							</resources>
+							<!--target -->
+							<outputDirectory>${target.root.dir}</outputDirectory>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+
+			<!-- Do replacements -->
+			<plugin>
+				<groupId>com.google.code.maven-replacer-plugin</groupId>
+				<artifactId>replacer</artifactId>
+				<version>1.5.2</version>
+				<executions>
+					<execution>
+						<id>replace</id>
+						<phase>process-resources</phase>
+						<goals>
+							<goal>replace</goal>
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<basedir>${target.root.dir}</basedir>
+					<includes>
+						<include>apiproxy/product-recommendations-v1.xml</include> <!-- PROJECT_ID, DESCRIPTION -->
+						<include>apiproxy/targets/default.xml</include> <!-- PROJECT_ID -->
+						<include>apiproxy/policies/AM-process-request-params.xml</include> <!-- userId value 6929470170340317899-1 -->
+						<include>apiproxy/policies/AM-create-query-request.xml</include> <!-- PROJECT_ID -->
+						<include>apiproxy/policies/SC-spanner-session.xml</include> <!-- PROJECT_ID -->
+						<include>test/integration/test-config.json</include> <!-- API_NORTHBOUND_DOMAIN, APIGEE_ENVIRONMENT -->
+						<include>edge.json</include> <!-- APIGEE_ENVIRONMENT -->
+					</includes>
+					<replacements>
+						<replacement>
+							<token>PROJECT_ID</token>
+							<value>${gcp.projectid}</value>
+						</replacement>
+						<replacement>
+							<token>API_NORTHBOUND_DOMAIN</token>
+							<value>${api.northbound.domain}</value>
+						</replacement>
+						<replacement>
+							<token>DESCRIPTION</token>
+							<value>commit ${commit} from branch ${branch} by ${user.name}</value>
+						</replacement>
+						<replacement>
+							<token>APIGEE_ENVIRONMENT</token>
+							<value>${apigee.env}</value>
+						</replacement>
+						<replacement>
+							<!-- CUSTOMER_USERID 6929470170340317899-1 in AM-process-request-params.xml replace with command line option -->
```

Comment:
can we define a better placeholder value

Reply:
Changed to use CUSTOMER_USERID